components/src/dynamo/planner/monitoring/traffic_metrics.py

```
@@ -282,21 +282,48 @@ def get_avg_request_duration(self, interval: str, model_name: str):
-                        f"{router_req_total}, use 0 instead"
+                        "No Prometheus metric data available for %s; using 0",
+                        router_requests_total,
                     )
                     return 0
-                value = float(result[0]["value"][1])
-                return 0 if math.isnan(value) else value
+                router_completed_count = float(completed_result[0]["value"][1])
+                return (
+                    0 if math.isnan(router_completed_count) else router_completed_count
+                )
             except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)